File: strategy/ducks/Fly/FlyWithWings.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace ducks.Fly
{
	public class FlyWithWings : IFlyBehaviuor
    {
		public FlyWithWings(StringBuilder stringBuilder) => _stringBuilder = stringBuilder;

		public void Fly() => _stringBuilder?.Append("fly with wings");

		private readonly StringBuilder _stringBuilder;
	}
}

File: strategy_java/src/ducks/strategies/FlyWithWings.java
package ducks.strategies;

import ducks.base.IFlyBehavior;

public class FlyWithWings implements IFlyBehavior {
    public FlyWithWings(StringBuilder sb) {
        _sb = sb;
    }

    @Override
    public void Fly() {
        _sb.append("with wings");
    }

    private final StringBuilder _sb;
}
